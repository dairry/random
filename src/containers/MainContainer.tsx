import React, { FunctionComponent, useState } from 'react';
import FormContainer from '~containers/FormContainer';
import Header from '~components/Header';
import SlotsContainer from '~containers/SlotsContainer';
import Result from '~components/Result';

type LimitsObj = {
  min: string | number;
  max: string | number;
};

const MainContainer: FunctionComponent = () => {
  const [amount, setAmount] = useState<number>(1);
  const [limits, setLimits] = useState<LimitsObj | null>(null);
  const [results, setResults] = useState<Array<number> | null>(null);
  const valueToAmount = (value: string | number) => {
    let val = value.toString().length;
    val = val < 1 ? 1 : val;

    if (val !== amount) {
      setLimits(null);
      setAmount(val);
    }
  };

  const doGenerate = ({ min, max }: LimitsObj) => setLimits({ min, max });
  const doClear = () => {
    setLimits(null);
    setAmount(1);
  };

  return (
    <div className="container">
      <Header />
      <Result results={results} />
      <SlotsContainer
        amount={!limits ? 1 : amount}
        limits={limits}
        onSetResults={setResults}
      />
      <FormContainer
        onSetMaxValue={valueToAmount}
        onGenerate={doGenerate}
        onClear={doClear}
      />
    </div>
  );
};

export default MainContainer;
