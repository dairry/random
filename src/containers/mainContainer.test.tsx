import React from 'react';
import { render } from '@testing-library/react';
import MainContainer from '~containers/MainContainer';

jest.mock('../utils', () => ({
  getRandomInt: () => 42,
}));

jest.mock('../hooks', () => ({
  useDebounce: func => func,
}));

describe('MainContainer component', () => {
  it('MainContainer should take a snapshot', () => {
    const { asFragment } = render(<MainContainer />);

    expect(asFragment()).toMatchSnapshot('MainContainer');
  });
});
