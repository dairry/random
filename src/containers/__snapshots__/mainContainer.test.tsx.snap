// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainContainer component MainContainer should take a snapshot: MainContainer 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <h1>
      Random number generator 
      <sup
        class="warning tooltip"
        data-tooltip="Generator of random positive integers in a given range"
      >
        *
      </sup>
    </h1>
    <div
      class="container container--flex"
    >
      <div
        class="slot-wrapper"
      >
        <div
          class="slot"
        >
          <div
            class="slot__cell"
          >
            0
          </div>
          <div
            class="slot__cell"
          >
            1
          </div>
          <div
            class="slot__cell"
          >
            2
          </div>
          <div
            class="slot__cell"
          >
            3
          </div>
          <div
            class="slot__cell"
          >
            4
          </div>
          <div
            class="slot__cell"
          >
            5
          </div>
          <div
            class="slot__cell"
          >
            6
          </div>
          <div
            class="slot__cell"
          >
            7
          </div>
          <div
            class="slot__cell"
          >
            8
          </div>
          <div
            class="slot__cell"
          >
            9
          </div>
        </div>
      </div>
    </div>
    <form>
      <div
        class="container container--flex container--flex-mobile"
      >
        <div
          class="form-group"
        >
          <label
            class="form-label"
            for="From"
          >
            From
          </label>
          <input
            class="form-input"
            id="From"
            min="0"
            placeholder="Min value"
            required=""
            type="number"
          />
        </div>
        <div
          class="form-group"
        >
          <label
            class="form-label"
            for="To"
          >
            To
          </label>
          <input
            class="form-input"
            disabled=""
            id="To"
            placeholder="Max value"
            required=""
            title="Add min value first"
            type="number"
          />
        </div>
      </div>
      <div
        class="container container--flex"
      >
        <button
          disabled=""
          title="Add limits first"
          type="submit"
        >
          Generate !
        </button>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
